sygnal/apnspushkin.py

```diff
@@ -237,7 +254,9 @@ async def _dispatch_request(self, log, span, device, shaved_payload, prio):
                         f"{response.status} {response.description}"
                     )
 
-    async def _dispatch_notification_unlimited(self, n, device, context):
+    async def _dispatch_notification_unlimited(
+        self, n: Notification, device: Device, context: NotificationContext
+    ) -> Optional[list]:
```

Comment:
```suggestion
    ) -> List[str]:
```
in general, avoid writing bare `list` as a type, and also try to specify the parameters if you can :)

Reply:
to elaborate, `list` really means `list[Any]` to mypy. There's an implicit hidden `Any` in there.

Reply:
^ further on that, the spelling is `List` with a capital `L`. I'm sure there's some strange historical reason at play :/

Reply:
I wouldn't say it's that strange. `list[...]` wasn't legitimate syntax until Python 3.9, because the `list` object didn't have a `__getitem__` defined. That changed with https://www.python.org/dev/peps/pep-0585/ . The PEP suggests that you can get this in 3.7 with `from __future__ import annotations`. In any case, we support 3.6 so must continue to use the old syntax. (We can get away with using the newer syntax in stub files, as long as mypy is sufficiently new.)